src/target/target_internal.h

```diff
@@ -103,12 +103,11 @@ struct target_command {
 typedef struct breakwatch breakwatch_s;
 
 struct breakwatch {
-	/* XXX: This needs adjusting for 64-bit operations */
 	breakwatch_s *next;
-	target_breakwatch_e type;
-	target_addr32_t addr;
+	target_addr64_t addr;
```

Comment:
This reordering needs to be gated on `sizeof(void *)` - for the firmware, pointers are 32-bit so this mis-aligns the 64-bit int of `addr`, while this correctly aligns it on 64-bit platforms. Conditionally reordering the `addr` and `type` members is probably the right way forwards.